tx-pool/src/process.rs

```diff
@@ -1053,46 +1028,41 @@ fn _update_tx_pool_for_reorg(
     // pending ---> gap ----> proposed
     // try move gap to proposed
     if mine_mode {
-        let mut entries = Vec::new();
+        let mut proposals = Vec::new();
         let mut gaps = Vec::new();
 
-        tx_pool.gap.remove_entries_by_filter(|id, tx_entry| {
-            if snapshot.proposals().contains_proposed(id) {
-                entries.push(tx_entry.clone());
-                true
-            } else {
-                false
+        for entry in tx_pool.pool_map.entries.get_by_status(&Status::Gap) {
+            let e = &entry.inner;
+            let short_id = e.proposal_short_id();
+            if snapshot.proposals().contains_proposed(&short_id) {
+                proposals.push(e.clone());
             }
-        });
-
-        tx_pool.pending.remove_entries_by_filter(|id, tx_entry| {
-            if snapshot.proposals().contains_proposed(id) {
-                entries.push(tx_entry.clone());
-                true
-            } else if snapshot.proposals().contains_gap(id) {
-                gaps.push(tx_entry.clone());
-                true
-            } else {
-                false
+        }
+
+        for entry in tx_pool.pool_map.entries.get_by_status(&Status::Pending) {
+            let e = &entry.inner;
+            let short_id = e.proposal_short_id();
+            if snapshot.proposals().contains_proposed(&short_id) {
+                proposals.push(e.clone());
+            } else if snapshot.proposals().contains_gap(&short_id) {
+                gaps.push(e.clone());
```

Comment:
As mentioned above, instead of passing the entry, you can just pass the short_id